src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,153 @@
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
+
+    if (editorState === EditorState.Selecting) {
+      toggleGroupRef?.current?.focus();
+    } else if (editorState === EditorState.Editing) {
+      inputRef?.current?.focus();
+    }
+  }, [editorState, setIsEditing]);
+
+  const onOpenChange = (isOpen: boolean) => {
+    if (!isOpen) {
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  const onCancel = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippageInputValue(percentSlippage.toString());
+  };
+
+  const onChangeSlippage = ({ value }: NumberFormatValues) => {
+    setSlippageInputValue(value);
+  };
+
+  const onConfirmSlippage = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippage(Number(slippageInputValue) / 100);
+  };
+
+  const onSelectSlippage = (newSlippage: string) => {
+    if (newSlippage === 'custom') {
+      setEditorState(EditorState.Editing);
+    } else {
+      setSlippage(Number(newSlippage));
+
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  if (disabled) {
+    return <Output type={OutputType.Percent} value={slippage} />;
+  }
+
+  return (
+    <WithConfirmationPopover
+      open={editorState !== EditorState.Viewing}
+      onOpenChange={onOpenChange}
+      align="end"
+      sideOffset={-22}
+      asChild
+      onCancel={onCancel}
+      onConfirm={editorState === EditorState.Editing ? onConfirmSlippage : undefined}
+      slotTrigger={
+        <button
+          onClick={() => setEditorState(EditorState.Selecting)}
+          tw="row gap-[0.5ch] underline"
+          type="button"
+          aria-label="edit slippage"
+        >
+          <Output type={OutputType.Percent} value={slippage} />
+          <Icon iconName={IconName.Pencil} />
+        </button>
+      }
+      tw="w-10 text-[0.625rem]"
+    >
+      {
+        {
+          [EditorState.Viewing]: undefined,
+          [EditorState.Selecting]: (
+            <div tw="inlineRow justify-center rounded-[0.5em] bg-color-layer-2">
+              <ToggleGroup
+                ref={toggleGroupRef}
+                items={[
+                  { label: '0.1%', value: '0.001' },
+                  { label: '0.25%', value: '0.0025' },
+                  { label: stringGetter({ key: STRING_KEYS.CUSTOM }), value: 'custom' },
+                ]}
```

Comment:
I think it makes sense to pull this out of the component for easier editing. Either within a `constants` file or at the top of this file

Reply:
just because we need to use stringGetter how about we move it to the top of the component? 

alternatively i can make it a custom hook but it just feels a bit overkill to do so

Reply:
sounds good!